src/js/InAppMenu.js

```diff
@@ -1,18 +1,24 @@
 import React from 'react';
 
 import AppHeader from './containers/Header';
-import Menu from './containers/Menu';
+import { VMenu, HMenu } from './containers/Menu';
+import store from './store'
 
 export default class InAppMenu extends React.Component {
     constructor() {
         super();
     }
 
     render() {
+        const state = store.getState()
+        const app = state.ui[state.activeApp]
+        var layout = app ? (app.activeSubMenu ? app.menu.find(x => x.menuID === app.activeSubMenu).menuLayout : app.menuLayout) : "LIST";
```

Comment:
Submenu souldnt have any impact on the top level app menu. If there is an active submenu, then the router would have the app in inAppList, not inAppMenu

Reply:
That makes sens. Removed the submenu logic in https://github.com/smartdevicelink/generic_hmi/pull/179/commits/a51b2fb95d7f7620d01a9566135817583345a4d1